staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -545,6 +551,114 @@ message ListPodSandboxResponse {
+    // ID of the pod for which to retrieve stats.
+    string pod_sandbox_id = 1;
+}
+
+message PodSandboxStatsResponse {
+    PodSandboxStats stats = 1;
+}
+
+// PodSandboxStatsFilter is used to filter containers.
+// All those fields are combined with 'AND'
+message PodSandboxStatsFilter {
+    // ID of the container.
+    string id = 1;
+    // LabelSelector to select matches.
+    // Only api.MatchLabels is supported for now and the requirements
+    // are ANDed. MatchExpressions is not supported yet.
+    map<string, string> label_selector = 2;
+}
+
+message ListPodSandboxStatsRequest {
+    // Filter for the list request.
+    PodSandboxStatsFilter filter = 1;
+}
+
+message ListPodSandboxStatsResponse {
+    // Stats of the pod.
+    repeated PodSandboxStats stats = 1;
+}
+
+// PodSandboxAttributes provides basic information of the container.
+message PodSandboxAttributes {
+    // ID of the pod.
+    string id = 1;
+    // Metadata of the pod.
+    PodSandboxMetadata metadata = 2;
+    // Key-value pairs that may be used to scope and select individual resources.
+    map<string,string> labels = 3;
+    // Unstructured key-value map holding arbitrary metadata.
+    // Annotations MUST NOT be altered by the runtime; the value of this field
+    // MUST be identical to that of the corresponding PodSandboxStatus used to
+    // instantiate the PodSandbox this status represents.
+    map<string,string> annotations = 4;
+}
+
+// PodSandboxStats provides the resource usage statistics for a pod.
+// The linux or windows field will be populated depending on the platform.
+message PodSandboxStats {
+    // Information of the pod.
+    PodSandboxAttributes attributes = 1;
+    // Stats from linux.
+    LinuxPodSandboxStats linux = 2;
+    // Stats from windows.
+    WindowsPodSandboxStats windows = 3;
+}
+
+// LinuxPodSandboxStats provides the resource usage statistics for a pod on linux
+message LinuxPodSandboxStats {
+    // CPU usage gathered from the pod.
+    CpuUsage cpu = 1;
+    // Memory usage gathered from the pod.
+    MemoryUsage memory = 2;
+    // Stats pertaining to CPU resources consumed by pod cgroup (which includes all containers' resource usage and pod overhead).
+    NetworkStats network = 3;
+    // Stats pertaining to processes in the pod.
+    ProcessStats process = 4;
+    // Stats of containers in the measured pod.
+    repeated ContainerStats containers = 5;
+}
+
+// WindowsPodSandboxStats provides the resource usage statistics for a pod on windows
+message WindowsPodSandboxStats {
+    // TODO: Add stats relevant to windows
+}
+
+// NetworkStats contains data about network resources.
+message NetworkStats {
+    // The time at which these stats were updated.
+    int64 timestamp = 1;
+
+    // Stats for the default interface, if found
+    InterfaceStats default_interface = 2;
+
+    repeated InterfaceStats interfaces = 3;
+}
+
+// InterfaceStats contains resource value data about interface.
+message InterfaceStats {
```

Comment:
Any chance this can be renamed to `NetworkInterfaceStats` or similar? "Interface" is a pretty overloaded term and I had to check the context before I understood this meant network interface.

Would require a KEP update.

Reply:
okay I have added this update to the kep https://github.com/kubernetes/enhancements/pull/2812
there is another naming inconsistency I found too, both of which I'll address here too.